niceman/distributions/base.py

```diff
@@ -40,6 +40,108 @@ def TypedList(type_):
 #
 
 class SpecObject(object):
+
+    # Empty tuple would signal that all attribs will be used while establishing
+    # the "identity" of the specobject
+    _cmp_fields = tuple()
+    # Fields of the primary interest when showing diff
+    _diff_fields = tuple()
+
+    @property
+    def _cmp_id(self):
+        if not self._cmp_fields:
```

Comment:
Hmm, but an empty tuple would go down this path.  That seems to contradict above which says that an empty tuple means "all attribs establish identity".

Reply:
I'm not sure I like the semantics of an empty _cmp_fields meaning use all of the attributes.  I'd rather require classes to define _cmp_fields explicitly and let the error be raised if they aren't.  This will mean one less special case we have to keep track of ("nothing means everything") and one more defense against a forgetful coder not thinking about _cmp_fields in a new class.

Reply:
@chaselgrove Sounds reasonable to me.  I've updated the comment to resolve the comment/code mismatch.

Reply:
Thanks.  Also removed the fallback to all attributes in the code.

Reply:
Ah, thanks, I hadn't spotted that.  So there was originally a code/comment mismatch *and* a code/code mismatch (the fallback was inaccessible due to the `not self._cmp_fields` guard).